SamarthTech/web-projects-2024
# Solve issue #46: Add Text To Speech Generator in React

### **Issue Solved:**
This pull request addresses and resolves **Issue #46**, which focuses on adding a Text-to-Speech (TTS) generator to the React application.

### **Features Implemented:**

- **Change Language Feature:**
  - Users can now select from multiple languages to synthesize text in their preferred language.
  - Available languages include English, Spanish, French, Russian, and more.
  
- **Change Voice Feature:**
  - Users can choose from different voices for each language, offering flexibility in how the speech is generated.



### **How to Test:**
1. Clone the repository and switch to the `textToSpeech` branch.
2. Run `npm install` to install dependencies.
3. Start the development server with `npm start`.
4. Navigate to `http://localhost:3000` and test the new Text-to-Speech features:
   - Enter text, choose a language, select a voice, and click the Speak button.
   - Test the download functionality to ensure the audio file is saved as an MP3.

### **Technologies Used:**
- React
- TypeScript
- TailwindCSS (for styling)

---

This pull request introduces a fully functional text-to-speech generator with a flexible language and voice selection feature, enhancing the application's interactivity.
